### ISSUE TITLE
React Components should support className

### ISSUE THREAD

**Issue by tonypee** [ionic-team/ionic#17962]
```
"@ionic/react": "^0.0.5",
"typescript": "^3.3.3"
"@types/react": "^16.8.4",
```

There is a bug with using 'className' on most components (IonCard,IonButton, Ion...)
```
<IonCard  className="myCard">...</IonCard>

errors:
Property 'className' does not exist on type 'IntrinsicAttributes & Pick<IonCardAttributes & { ref?: RefObject<HTMLIonCardElement>; children?: ReactNode; }, "hidden" | "dir" | "slot" | "style" | "title" | "contextmenu" | ... 157 more ... | "onTransitionEndCapture"> & RefAttributes<...>'.ts(2322)
```

**Comment by tonypee** [comment]
It seems that the root of this, is in the stencil library where the definition of JSX.HTMLAttributes only allows 'class' not 'className'. 

React officially supports **both** className and class... I guess you have decided 'class' is the way of the future, or more cross platform (which is true). It may be a stumbling block for many react developers however.

**solution: USE CLASS**
```
<IonCard  class="myCard">...</IonCard>
```

**Comment by tonypee** [comment]
Since React doesn't yet support class, would it make sense to allow className on that platform? Otherwise im mixing the two attributes depending on the element

**Comment by tonypee** [comment]
Im finding that styling the IonCard at all (via class or className (typing as any)) produces unexpected renders. My dom is rendering visibilities incorrectly. Maybe the way class integrates with react currently is unsupported, and therefore stencil's approach is dangerous. more testing is needed

**Comment by SulimanTwfiq** [comment]
[i have the same issue](https://forum.ionicframework.com/t/the-component-disappear-when-i-put-classname-for-css-styling-ionic-react/160937)

**Comment by jthoms1** [comment]
I am going to create some tests around being about support className and css in js solutions.  Leaving this open as a bug to ensure I have a narrative on how they can be used.

**Comment by elylucas** [comment]
Fixed in #18231. Thanks!
